### PR TITLE
ostree: Remove jQuery usage from OSTree code

### PR DESCRIPTION
**Pull request by stefwalter**
This is prior to migrating to webpack so we have to pull in
less dependencies here.

### CODE REVIEW

File: pkg/ostree/app.js [hunk 1]
```diff
@@ -163,23 +162,24 @@ require([
...
                     done(function () {
-                        $(client).on("changed.ostreeIndex", function() {
-                            $scope.$applyAsync(function() {
-                                $scope.runningMethod = client.running_method;
-                                $scope.os_list = client.os_list;
-                            });
-                        });
-
+                        client.addEventListener("changed", on_changed);
                         $scope.$applyAsync(function() {
```
Comment (petervo) [inline pkg/ostree/app.js]: Probably just want to call on_changed here.

**Comment by petervo** [comment]
Unfortunately rpm-ostree is broken in atomic right now. Let me see if I can get a centos-continuous image going so that we can see the tests run properly.

**Comment by petervo** [comment]
Body doesn't seem to be loading correctly.

**Comment by petervo** [comment]
Opened #4768 for testing the centos-continuous image but the fix for rpm-ostree isn't in yet.

File: pkg/ostree/app.js (hunk 1 above)
Comment (stefwalter) [inline pkg/ostree/app.js]: Done.

**Comment by stefwalter** [comment]
@petervo I agree this should block on all the ostree tests passing.

**Comment by stefwalter** [comment]
Fixed several cases.

File: pkg/ostree/client.js [hunk 2]
```diff
@@ -448,13 +457,13 @@ define([
...
-            var dp = $.Deferred();
+            var dp = cockpit.defer();
             var i;
             var reboot = false;
 
-            if ($.isArray(method_args)) {
+            if (Array.isArray(method_args)) {
                 for (i = 0; i < method_args.length; i++) {
-                    if ($.isPlainObject(method_args[i])) {
+                    if ("reboot" in method_args[i]) {
```
Comment (petervo) [inline pkg/ostree/client.js]: This isn't quite the same thing and causes and error when method_args[i] is a string.

**Comment by petervo** [comment]
Replaced by #4799

**Pull request closed without merging**